src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -2995,6 +3048,39 @@ public final boolean equals(Object o)
         }
     }
 
+    public static class ConsensusAttemptResult
```

Comment:
wdyt about having a single private ctor, and having 3 static construction methods that named something like `txnResult`, `paxosResult`, and `switchProtocol`, or something like that? At least in the case of `retry_new_protocol`, it would make it a bit easier to understand what's happening without having to remember how the various ctor invocations affect it's state